src/animals/animals.controller.ts

```diff
@@ -0,0 +1,28 @@
+import { Controller, Post, Body, Get, Delete, Param } from '@nestjs/common';
+import { AnimalsService } from './animals.service';
+import { CreateAnimalDto } from './dto/create-animal.dto';
+
+@Controller('animals')
+export class AnimalsController {
+  constructor(private readonly animalService: AnimalsService) {}
+
+  @Post('')
+  async create(@Body() createAnimalDto: CreateAnimalDto) {
+    return await this.animalService.create(createAnimalDto);
+  }
+
+  @Get('')
+  async findAll() {
+    return await this.animalService.findAll();
```

Comment:
If you return a promise, no need to await it during `return`
```suggestion
    return this.animalService.findAll();
```